pkg/queue/prometheus_stats_reporter_test.go

```diff
@@ -160,25 +165,24 @@ func TestProtobufStatsReporterReport(t *testing.T) {
 				RequestCount:              test.reqCount,
 				ProxiedRequestCount:       test.proxiedReqCount,
 			})
-			checkData(t, requestsPerSecondGV, test.expectedReqCount)
-			checkData(t, averageConcurrentRequestsGV, test.expectedConcurrency)
-			checkData(t, proxiedRequestsPerSecondGV, test.expectedProxiedRequestCount)
-			checkData(t, averageProxiedConcurrentRequestsGV, test.expectedProxiedConcurrency)
-
-			if got := getData(t, processUptimeGV); got < 5.0 || got > 6.0 {
-				t.Errorf("Got %v for process uptime, wanted 5.0 <= x < 6.0", got)
+			want := metrics.Stat{
+				RequestCount:                     test.expectedReqCount,
+				AverageConcurrentRequests:        test.expectedConcurrency,
+				ProxiedRequestCount:              test.expectedProxiedRequestCount,
+				AverageProxiedConcurrentRequests: test.expectedProxiedConcurrency,
```

Comment:
Can we now just specify `in Stat` and `expect Stat` in the test rather than spelling out the individual fields maybe?

Reply:
@markusthoemmes could you explain more what you mean here?

Reply:
Well we mostly copy in fields from the test struct into the respective input/output structs of the test.

Can we instead do

```golang
tests := []struct[
  ...
  in network.RequestStatsReport
  want Stat
```

and then use those in the tests directly?

Reply:
will check.